test/jit/test_slice.py

```diff
@@ -136,3 +136,25 @@ def tuple_slice(a):
         self.assertTrue(num_outputs == {2})
         self.run_pass('lower_all_tuples', tuple_graph)
         self.assertTrue('Tuple' not in str(tuple_graph))
+
+    def test_module_list_slicing(self):
+        class Foo(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                m1 = torch.nn.Linear(1, 2)
+                m2 = torch.nn.ReLU()
+                m3 = torch.nn.GRU(1, 2)
+                self.test = torch.nn.ModuleList([m1, m2, m3])
+
+            def forward(self):
+                return self.test[::-2], self.test[1:2:]
+
+        scripted_foo = torch.jit.script(Foo())
+        result1, result2 = scripted_foo()
+
+        self.assertEqual(len(result1), 2)
+        self.assertEqual(str(result1[0]._type()), "__torch__.torch.nn.modules.rnn.GRU")
+        self.assertEqual(str(result1[1]._type()), "__torch__.torch.nn.modules.linear.Linear")
+
+        self.assertEqual(len(result2), 1)
+        self.assertEqual(str(result2[0]._type()), "__torch__.torch.nn.modules.activation.ReLU")
```

Comment:
Type string comparison is fragile, please use a more robust method to verify test results.

One potential solution is to create a dummy module class that has a string-typed attribute. Then you can create 3 instances that each has a different value for that attribute. This is both lighter-weight and more robust than current implementation.

Reply:
done